sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -1158,6 +1158,7 @@ object TableFunctionRegistry {
     generator[PosExplode]("posexplode"),
     generator[PosExplode]("posexplode_outer", outer = true),
     generator[Stack]("stack"),
+    generator[AllCollations]("all_collations"),
```

Comment:
given we have `sql_keywords`, shall we call it `string_collations`?

Reply:
Okay

Reply:
nit: To me this sounds weird. We suggest that there might exist collations for some other type, for keywords it is more likely to have other keywords (python, scala ...) @cloud-fan

Reply:
oh, not only string type? then `all_collations` is good

Reply:
Okay, Let's restore to `all_collations`.

Reply:
all_collation is fine although I'm not sure why all is necessary. But no strong feelings.

Reply:
It's `all_collation`, not `all_collations`, right?